pkg/controller/nicclusterpolicy/nicclusterpolicy_controller.go

```diff
@@ -124,6 +132,19 @@ func (r *ReconcileNicClusterPolicy) Reconcile(request reconcile.Request) (reconc
 		return reconcile.Result{}, err
 	}
 
+	if !isSupportedInstance {
+		instance.Status.State = mellanoxv1alpha1.StateIgnore
```

Comment:
can you put all the logic in a helper method : `handleUnsupportedInstance()` ?
also the Logging from L#113-119